File: SettlementDoneEasy/Models/Document/DocumentModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SDE_Server.Models.Document
{
    public class DocumentModel
    {
        public int ID { get; set; }
        public int? UserID { get; set; }
        public string Data { get; set; }
        public int? TemplateID { get; set; } // idk if this will be needed so i'll leave it in for now
        public DocumentDataModel DocumentData { get; set; }
    }
}

File: SettlementDoneEasy/ClientApp/src/app/features/document-management/models/DocumentModel.ts
import { DocumentAuditModel } from "./DocumentAuditModel";
import { DocumentData } from "./DocumentData";
import { DocumentDataModel } from "./DocumentDataModel";
import { DocumentTemplateModel } from "./DocumentTemplateModel";

export class DocumentModel {
	public id: number;
	public userId: number;
	public data: any | DocumentData;
	public templateID: number;
	public creationDate: Date;
	public title: string;


	public template: DocumentTemplateModel;
	public audits: DocumentAuditModel[];
	public documentData: DocumentDataModel;
}


